src/ShellClients/WindowPositioner.vala

```diff
@@ -7,13 +7,28 @@
 
 public class Gala.WindowPositioner : Object {
     public enum Position {
-        CENTER
+        TOP,
+        BOTTOM,
+        CENTER;
+
+        public static Position from_anchor (Pantheon.Desktop.Anchor anchor) {
+            if (anchor > 1) {
+                warning ("Position %s not supported yet", anchor.to_string ());
+                return CENTER;
+            }
+
+            return (Position) anchor;
+        }
     }
 
     public Meta.Display display { get; construct; }
     public Meta.Window window { get; construct; }
-    public Position position { get; private set; }
-    public Variant? position_data { get; private set; }
+    /**
+     * This may only be set after the window was shown.
+     * The initial position should only be given in the constructor.
+     */
+    public Position position { get; construct set; }
+    public Variant? position_data { get; construct set; }
```

Comment:
I'm not sure what you mean here? The position passed to the constructor is used until a new one is set (which currently never happens, since we don't support changing panel positions)

Reply:
Ok sorry, I got confused because `position_window` is not called inside the constructor and I assumed it was only called when the position changed